### PR TITLE
ROX-21771: add cloud sources datastore

### PR DESCRIPTION
**Pull request by stehessel**
## Description

Add the datastore for the new cloud source integration type. See https://github.com/stackrox/stackrox/pull/9537 for context. A cloud source integration provides clusters to be discovered for on-boarding purposes.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI e2e test

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by stehessel** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/9542?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #9537** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/9537?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #9542** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/9542?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/9542?utm_source=stack-comment).

File: central/cloudsources/datastore/datastore_impl.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		return errors.New("empty cloud source")
+	}
+
+	errorList := errorhelpers.NewErrorList("Validation")
+	if stringutils.AtLeastOneEmpty(
+		cloudSource.GetId(),
+		cloudSource.GetName(),
+		cloudSource.GetCredentials().GetSecret(),
+	) {
+		errorList.AddString("all required fields must be set")
```
Comment (dhaus67) [inline central/cloudsources/datastore/datastore_impl.go:78]: Let's also include the field names?

File: central/cloudsources/datastore/datastore_impl_postgres_test.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	s.ErrorIs(err, errox.NotFound)
+	s.Empty(cloudSource)
+
+	cloudSource = fixtures.GetStorageCloudSource()
+	err = s.datastore.UpsertCloudSource(s.writeCtx, cloudSource)
+	s.Require().NoError(err)
+
+	roundtripCloudSource, err := s.datastore.GetCloudSource(s.readCtx, cloudSource.GetId())
+	s.NoError(err)
+	s.Assert().Equal(cloudSource, roundtripCloudSource)
```
Comment (dhaus67) [inline central/cloudsources/datastore/datastore_impl_postgres_test.go:80]: super nit: let's be consistent and either always use `Assert()` or the e.g. `s.Empty`.

**Review by dhaus67 — APPROVED** [review]
Two minor comments, otherwise LGTM.

File: central/cloudsources/datastore/datastore_impl.go (hunk 1 above)
Comment (stehessel) [reply central/cloudsources/datastore/datastore_impl.go:78]: refactored in next PR

**Pull request merged**